File: examples/vision-camera/modules/webrtc-source/ios/webrtcframeprocessor/WebrtcFrameProcessorPlugin.swift
import FishjamCloudClient
import VisionCamera

class WebrtcVisionCameraCustomSource: CustomSource {
  var delegate: CustomSourceDelegate?
  
  let isScreenShare = false
  let metadata = ["type":"camera"].toMetadata()
  let videoParameters = VideoParameters.presetFHD43
}

public class WebrtcFrameProcessorPlugin: FrameProcessorPlugin {
  static var currentSource: WebrtcVisionCameraCustomSource?

  public override func callback(_ frame: Frame, withArguments arguments: [AnyHashable : Any]?) -> Any {
    if let customSource = WebrtcFrameProcessorPlugin.currentSource {
      WebrtcFrameProcessorPlugin.currentSource?.delegate?.customSource(customSource, didOutputSampleBuffer: frame.buffer)
    }
    
    return frame
  }
}

File: examples/vision-camera/modules/webrtc-source/android/src/main/java/io/fishjam/example/webrtcsource/webrtcframeprocessor/WebrtcFrameProcessorPlugin.kt
package io.fishjam.example.webrtcsource.webrtcframeprocessor

import com.mrousavy.camera.frameprocessors.Frame
import com.mrousavy.camera.frameprocessors.FrameProcessorPlugin
import com.mrousavy.camera.frameprocessors.VisionCameraProxy
import io.fishjam.example.webrtcsource.WebrtcVisionCameraCustomSource

class WebrtcFrameProcessorPlugin(proxy: VisionCameraProxy, options: Map<String, Any>?): FrameProcessorPlugin() {
  companion object {
    var currentSource: WebrtcVisionCameraCustomSource? = null
  }

  override fun callback(frame: Frame, arguments: Map<String, Any>?): Frame {
    currentSource?.consumer?.onImageProxyCaptured(frame.imageProxy)
    return frame
  }
}
